Bump version of mountpoint-s3 to v1.0.2 (#530)
awslabs/mountpoint-s3: mountpoint-s3/CHANGELOG.md

## v1.0.2 (September 22, 2023)

### Breaking changes
* No breaking changes.

### Other changes
* New Mountpoint releases are built on CentOS 7 instead of Amazon Linux 2. This lowers the minimum requirement to run Mountpoint to glibc 2.17 or newer. ([#517](https://github.com/awslabs/mountpoint-s3/pull/517))

> * Updated the CRT submodules to the latest releases. This includes bug fixes in aws-c-s3 library such as aws-c-s3#347 and aws-c-s3#346. ([#529](https://github.com/awslabs/mountpoint-s3/pull/529))

Reviewer: Specifically for CRT updates, I'd lead with what's fixed rather than the fact we updated it. Customers don't care about dependency updates unless they bring some useful new thing, so it's the new thing we should describe rather than the CRT itself.

Reviewer: (If we think there's no useful new thing worth mentioning, then we should just drop it)

Reviewer 2: I think we should mention the fix in aws-c-s3#347: Fixed an issue with internal resource cleanup which could lead to Mountpoint hanging after a high number of file uploads.

* Fixed a bug where writing to a file for longer than five minutes will result in a panic. ([#513](https://github.com/awslabs/mountpoint-s3/pull/513))

> * Updated the prefetcher to cancel discarded tasks and free up some unused resources. ([#505](https://github.com/awslabs/mountpoint-s3/pull/505))

Reviewer: ...for random read workloads

> * Updated the CRT submodules to the latest releases. This includes fixing an issue with internal resource cleanup which could lead to Mountpoint hanging after a high number of file uploads. ([#529](https://github.com/awslabs/mountpoint-s3/pull/529))

Reviewer 2: As @jamesbornholt suggested, should we just leave out the CRT update and focus on the fix? E.g.
```suggestion
* Fixed an issue with internal resource cleanup which could lead to Mountpoint hanging after a high number of file uploads. ([#529](https://github.com/awslabs/mountpoint-s3/pull/529))
```

## v1.0.2 (September 22, 2023)

### Breaking changes
* No breaking changes.

### Other changes

> * Building the Mountpoint release on CentOS 7. This gets us compatibilty back to glibc 2.17. This should let mountpoint run on OS which by default packages GCC and Clang that were too old to build CRT and run bindgen respectively in previous release. ([#517](https://github.com/awslabs/mountpoint-s3/pull/517))

Reviewer 3: ```suggestion
* New Mountpoint releases are now built on CentOS 7 instead of Amazon Linux 2. This lowers minimum requirement to run Mountpoint to require glibc 2.17 or newer. ([#517](https://github.com/awslabs/mountpoint-s3/pull/517))
```

Reviewer 2: nits: 
* remove "now"
* requirement/to require sounds redundant

How about:
```suggestion
* New Mountpoint releases are built on CentOS 7 instead of Amazon Linux 2. This lowers the minimum requirement to run Mountpoint to glibc 2.17 or newer. ([#517](https://github.com/awslabs/mountpoint-s3/pull/517))
```

> * Updated the CRT submodules to the latest release which is one of the mountpoint-s3 dependencies. This majorly resolves the issue of mountpoint-s3 freeze when trying to upload files to s3 without knowing their full size at the time of upload and there are multiple such file to take time. ([#529](https://github.com/awslabs/mountpoint-s3/pull/529))

Reviewer 3: Updated the CRT submodules to the latest releases. This includes bug fixes in `aws-c-s3` library such as [aws-c-s3#347](https://github.com/awslabs/aws-c-s3/pull/347) and [aws-c-s3#346](https://github.com/awslabs/aws-c-s3/pull/346). ([#529](https://github.com/awslabs/mountpoint-s3/pull/529))

Author: But changelog should also include what has changed and why customers should care about as recieved review in the last upgrade.

Reviewer 3: For this, I think the main changes are in aws-c-s3 not mountpoint and we can't confirm that it will fix the freeze problem in mountpoint yet.

> * Updated the HistogramFn in metrics to log failure rather than panic when size of file is too large for histogram value. With this change mountpoint-s3 continues to operate after recieving histogram value too large. For example, earlier when writing 50GiB of data to a file, mountpoint-s3 panicked due to an unexpected "ValueOutOfRangeResizeDisabled" error. ([#513](https://github.com/awslabs/mountpoint-s3/pull/513))

Reviewer 3: ```suggestion
* Fixed a bug when writing to a file in excess of five minutes will result in a panic. ([#513](https://github.com/awslabs/mountpoint-s3/pull/513))
```

Reviewer 2: I'd replace "in excess of" with "for longer than"

Reviewer 3: The real problem for customer is the panic on writing. It's true that histogram function is the root cause, but it's probably going too much into details since we also have to link to the PR.

> * Updated the prefetcher to cancel discarded tasks. With this change, prefetcher cancel unused GET requests, freeing up bandwidth for GETs that will be used to fulfill other mountpoint-s3 reads. ([#505](https://github.com/awslabs/mountpoint-s3/pull/505))

Reviewer 3: ```suggestion
* Updated the prefetcher to cancel discarded tasks and free up some unused resources. ([#505](https://github.com/awslabs/mountpoint-s3/pull/505))
```

Author: Again here, not sure if making the overview less descriptive helps. Can go ahead with the suggested change.

Reviewer 3: The message in the PR is misleading a bit at first because we didn't cancel the inflight GETs. so, freeing up bandwidth is kind of not true. We only free up some memory in the prefetcher.
